File: DSA/RATNA_KHANNA_590013141/DAY_12_QUESTION_1.c
#include <stdio.h>
#include <string.h>

int isRotation(char str1[], char str2[]) 
{
    int len1 = strlen(str1);
    int len2 = strlen(str2);

    if (len1 != len2) 
    {
        return 0; 
    }

    char temp[200]; 
    int i, j, k;
    
    for (i = 0; i < len1; i++) {
        temp[i] = str1[i];
    }
    for (j = 0; j < len1; j++) {
        temp[i + j] = str1[j];
    }
    temp[i + j] = '\0'; 
    for (i = 0; i <= 2 * len1 - len2; i++) 
    {
        for (j = 0; j < len2; j++) {
            if (temp[i + j] != str2[j]) {
                break;
            }
        }
        if (j == len2) {
            return 1; 
        }
    }
    return 0; 
}

int main() {
    char str1[100], str2[100];

    printf("Enter first string: ");
    scanf("%s", str1);

    printf("Enter second string: ");
    scanf("%s", str2);

    if (isRotation(str1, str2)) {
        printf("True\n");
    } else {
        printf("False\n");
    }

    return 0;
}

File: DSA/Prabhjeet_590013286/Day12/Day_12_Question_1.py
# Problem 1: String Rotation
# Given two strings, determine whether one string 
# is a rotation of the other. A rotation means characters 
# can be shifted from the front to the end.
# Efficiently solve this without generating all possible rotations.
# Example: "abcde" and "cdeab" → True
def is_rotation(s1, s2):
    return len(s1) == len(s2) and s2 in (s1 + s1)

str1 = input("Enter the first string: ")
str2 = input("Enter the second string: ")

if is_rotation(str1, str2):
    print(" The second string is a rotation of the first.")
else:
    print(" The second string is NOT a rotation of the first.")
